src/gutter-container.coffee

```diff
@@ -0,0 +1,131 @@
+{Model} = require 'theorist'
+{CompositeDisposable, Emitter} = require 'event-kit'
+{Gutter} = require './gutter'
+
+# This class encapsulates the logic for adding and modifying the gutters on a
+# TextEditor.
+
+module.exports =
+class GutterContainer extends Model
+  # @param textEditor (required) The TextEditor that this GutterContainer is
+  #   associated with.
+  constructor: (textEditor) ->
+    @textEditor = textEditor
+    @leftGutters = []
+    @rightGutters = []
+    @defaultGutter = null
+    @emitter = new Emitter
+
+  destroy: ->
+    @leftGutters = null
+    @rightGutters = null
+    @defaultGutter = null
+    @emitter.dispose()
+    super()
+
+  # Adds a gutter to the left side of the line numbers. Returns a Gutter instance.
+  # @param gutterName (required) a unique string that identifies this gutter in
```

Comment:
We should probably use [atomdoc](https://github.com/atom/atomdoc) on all these docs.